### ISSUE TITLE
hold_peaks should convert input image to float

### ISSUE THREAD

**Issue by madeline-am** [PMEAL/porespy#350]
error if a boolean is passed, numpy reports can't subtract boolean images

**Comment by Ni2M** [comment]
I think the user should not pass boolean images as the input argument, as the function works with greyscale images. We could add more descriptions in function's docstring. However, for binary images as the input, the user can convert True,False to 1,0 using im.astype(int) before passing to the function. If more user-friendly options are desired, we could add a check at the beginning of the function to convert boolean inputs before proceeding. The application of hold_peaks might be more related to distance transform (which is greyscale). Here is a code to show the differences:
```python
import porespy as ps
import numpy as np
import matplotlib.pyplot as plt
import scipy.ndimage as spim
np.random.seed(10)
im = ps.generators.blobs([100,100],porosity=0.2,blobiness=0.8)
plt.figure()
plt.imshow(im)
plt.colorbar()
plt.title('binary_image')
dt = spim.distance_transform_edt(input=im) # input image is binary
plt.figure()
plt.imshow(dt)
plt.colorbar()
plt.title('dt')
im2= ps.filters.hold_peaks(im.astype(int),axis=0) # input image is greyscale
plt.figure()
plt.imshow(im2)
plt.colorbar()
plt.title('hold_peaks_on_void')
im3= ps.filters.hold_peaks(dt,axis=0) # input image is greyscale (probably common application)
plt.figure()
plt.imshow(im3)
plt.colorbar()
plt.title('hold_peaks_on_dt')
im4= ps.filters.find_peaks(dt,r_max=1) # input image is greyscale
plt.figure()
plt.imshow(im4)
plt.colorbar()
plt.title('find_peaks_on_dt')
```

To change the function so that boolean inputs will be accepted, line https://github.com/PMEAL/porespy/blob/ec0b97a53d1e9c2f2e6cfaef2c86cdd1bc6bb033/porespy/filters/__funcs__.py#L99 should be changed to : A = im.astype(float).

**Comment by Ni2M** [comment]
The issue is addressed in PR#415 and merged.
